File: src/main/java/org/openbase/planetsudo/level/save/MarioWorld.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package org.openbase.planetsudo.level.save;

/*-
 * #%L
 * PlanetSudo GameEngine
 * %%
 * Copyright (C) 2009 - 2019 openbase.org
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * #L%
 */

import java.awt.Color;
import java.awt.Polygon;
import org.openbase.planetsudo.geometry.Base2D;
import org.openbase.planetsudo.geometry.Direction2D;
import org.openbase.planetsudo.level.AbstractLevel;
import org.openbase.planetsudo.level.ResourcePlacement;
import org.openbase.planetsudo.level.ResourceRandomRectangle;
import org.openbase.planetsudo.level.levelobjects.Resource.ResourceType;

/**
 *
 * @author noxus
 */
public class MarioWorld extends AbstractLevel {

	@Override
	protected Polygon loadLevelBorderPolygon() {
		Polygon levelBorders = new Polygon();
		levelBorders.addPoint(0, 0);
		levelBorders.addPoint(500, 400);
		levelBorders.addPoint(1000, 0);
		levelBorders.addPoint(1000, 1000);
		levelBorders.addPoint(800, 1000);
		levelBorders.addPoint(800, 700);
		levelBorders.addPoint(500, 1000);
		levelBorders.addPoint(200, 700);
		levelBorders.addPoint(200, 1000);
		levelBorders.addPoint(0, 1000);
		return levelBorders;
	} 



	@Override
	protected Base2D[] loadHomePositions() {
		Base2D[] homePosition = new Base2D[2];
		homePosition[0] = new Base2D(100, 950, Direction2D.DOWN);
		homePosition[1] = new Base2D(900, 950, Direction2D.DOWN);
		return homePosition;
	}

	@Override
	protected ResourcePlacement[] loadResourcePlacement() {
		ResourcePlacement[] resourcePlacement = new ResourcePlacement[4];
		resourcePlacement[0] = new ResourceRandomRectangle(0, 0, 200, 500, 10, ResourceType.Normal);
		resourcePlacement[1] = new ResourceRandomRectangle(250, 300, 550, 550, 15, ResourceType.DoublePoints);
		resourcePlacement[2] = new ResourceRandomRectangle(200, 900, 400, 100, 15, ResourceType.ExtremPoint);
		resourcePlacement[3] = new ResourceRandomRectangle(800, 0, 200, 500, 10, ResourceType.Normal);
		return resourcePlacement;
	}

	@Override
	protected Color loadLevelColor() {
		return new Color(174,94,94);
	}

	@Override
	protected Polygon[] loadLevelWallPolygons() {
		return null;
	}
}

File: src/main/kotlin/org/openbase/planetsudo/level/save/MarioWorld.kt
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package org.openbase.planetsudo.level.save

import org.openbase.planetsudo.geometry.Base2D
import org.openbase.planetsudo.geometry.Direction2D
import org.openbase.planetsudo.level.AbstractLevel
import org.openbase.planetsudo.level.ResourceRandomRectangle
import org.openbase.planetsudo.level.levelobjects.Resource.ResourceType
import java.awt.Color
import java.awt.Polygon

/**
 *
 * @author noxus
 */
class MarioWorld : AbstractLevel() {
    override fun loadLevelBorderPolygon() = Polygon().apply {
        addPoint(0, 0)
        addPoint(500, 400)
        addPoint(1000, 0)
        addPoint(1000, 1000)
        addPoint(800, 1000)
        addPoint(800, 700)
        addPoint(500, 1000)
        addPoint(200, 700)
        addPoint(200, 1000)
        addPoint(0, 1000)
    }

    override fun loadHomePositions() = listOf(
        Base2D(100.0, 950.0, Direction2D.DOWN),
        Base2D(900.0, 950.0, Direction2D.DOWN),
    )

    override fun loadResourcePlacement() = listOf(
        ResourceRandomRectangle(0, 0, 200, 500, 10, ResourceType.Normal),
        ResourceRandomRectangle(250, 300, 550, 550, 15, ResourceType.DoublePoints),
        ResourceRandomRectangle(200, 900, 400, 100, 15, ResourceType.ExtremPoint),
        ResourceRandomRectangle(800, 0, 200, 500, 10, ResourceType.Normal),
        ResourceRandomRectangle(400,0,200,450,10,ResourceType.Tonic)
    )

    override fun loadLevelColor() = Color(174, 94, 94)

    override fun loadLevelWallPolygons() = emptyList<Polygon>()
}
